ppb/events.py

```diff
@@ -85,6 +89,30 @@ def __event__(self, bag, fire_event):
                     raise
 
 
+class TreeStructurePublisher(EventMixin, IterableABC):
+    """
+    A mixin to publish events to an iterable's children.
+    """
+
+    def __iter__(self):
```

Comment:
I'm not super stoked about reserving `__iter__` to define children. I would rather see a `__children__` or `__tree_children__`.

I know we don't use it for anything right now, but it's such a huge part of Python it seems like a bad idea to reserve it so no game object can use it for something more semantic.